src/controllers/userController.js

```diff
@@ -1,59 +1,97 @@
 const User = require('../models/User');
-const jwt = require('jsonwebtoken');
-
-// create token (we use it here after updating user's data )
-const createJWT = ({ payload }) => {
-  const token = jwt.sign(payload, process.env.JWT_SECRET, {
-    expiresIn: process.env.JWT_LIFETIME,
-  });
-  return token;
-};
+const { StatusCodes } = require('http-status-codes');
+const CustomError = require('../errors');
+const {
+  createTokenUser,
+  attachCookiesToResponse,
+  checkPermissions,
+} = require('../utils');
 
-//returning a regular javascript object that is a subset of the fields of the User model.
-const formatUserForFrontend = (user) => {
-  return { name: user.name, userId: user._id, role: user.role };
+const getAllUsers = async (req, res) => {
+  // Function to get all users
+  // Find all users with the role 'user' in the database and exclude the 'password' field
+  const users = await User.find({ role: 'user' }).select('-password');
+  res.status(StatusCodes.OK).json({ users }); // Send a JSON response with the status code 200 OK and the users
 };
 
-
 const getSingleUser = async (req, res) => {
-  const user = await User.findOne({ _id: req.params.id });
+  // Function to get a single user by ID
+  // Find the user in the database based on the provided user ID and exclude the 'password' field
+  const user = await User.findOne({ _id: req.params.id }).select('-password');
   if (!user) {
-    res.status(400).json({ msg: `No user with id : ${req.params.id}` });
+    // Throw a NotFoundError if the user is not found
+    throw new CustomError.NotFoundError(`No user with id: ${req.params.id}`);
   }
-  // If the user is found, it calls the formatUserForFrontend function to format the user data before sending it in the response
-  const tokenUser = formatUserForFrontend(user);
+  checkPermissions(req.user, user._id); // Check if the user has permission to access the user's information
+  res.status(StatusCodes.OK).json({ user }); // Send a JSON response with the status code 200 OK and the user
+};
 
-  res.status(200).json({ user : tokenUser });
+const showCurrentUser = async (req, res) => {
+  try {
+    const user = await User.findOne({ _id: req.user.userId }).select(
+      '-password'
+    );
+    if (!user) {
+      throw new CustomError.NotFoundError('User not found');
+    }
+    res.status(StatusCodes.OK).json({ user });
+  } catch (error) {
+    next(error);
+  }
```

Comment:
```suggestion
  const user = await User.findOne({ _id: req.user.userId }).select(
    '-password'
  );
  if (!user) {
    throw new CustomError.NotFoundError('User not found');
  }
  res.status(StatusCodes.OK).json({ user });
```

^ We don't need to wrap this in a `try/catch` statement because we're using `express-async-errors` library and have error handling middleware which will catch and handle the error and pass it along to the `next()` middleware if needed.

Reply:
<img width="512" alt="Снимок экрана 2023-08-10 170418" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/b0d24f7d-c914-4f4a-a681-ebbe5b420323">